kzg-bench/src/tests/eip_4844.rs

```diff
@@ -9,15 +9,16 @@ use kzg::eip_4844::{
     FIELD_ELEMENTS_PER_BLOB, TRUSTED_SETUP_PATH,
 };
 use kzg::{FFTSettings, Fr, KZGSettings, Poly, G1, G2};
+use pathdiff::diff_paths;
 use rand::rngs::ThreadRng;
 use rand::Rng;
-use std::env::set_current_dir;
+use std::env::current_dir;
 use std::fs;
 use std::path::PathBuf;
 
 fn u64_to_bytes(x: u64) -> [u8; 32] {
     let mut bytes = [0u8; 32];
-    bytes[0..8].copy_from_slice(&x.to_le_bytes());
+    bytes[24..32].copy_from_slice(&x.to_be_bytes());
     bytes
 }
```

Comment:
This change will affect other backends that expect from_bytes to be LE format. I'm not sure if this is relevant to mcl or zkcrypto, haven't checked, but it does affect arkworks.

Probably same thing is also in other modified tests, have we decided that all from_bytes should expect BE format?

Reply:
To clarify - I'm fine with changing arkworks implementation to convert from BE to LE, just think that from_bytes should be consistent across all backends.

Reply:
Other backends cannot expect bytes to be in LE format, as it is incorrect - by standard bytes must be present as BE ([see "constants" section "KZG_ENDIANNESS"](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#constants)).

So generally speaking, all `eip4844` functions *must* read bytes as BE, otherwise this is not correct behavior. Although some other functions, like those in [`Fr`](https://github.com/ArtiomTr/rust-kzg/blob/9482c91f7bb744c8d429bcdc08731282eb1937bc/kzg/src/lib.rs#L20-L30), [`G1`](https://github.com/ArtiomTr/rust-kzg/blob/9482c91f7bb744c8d429bcdc08731282eb1937bc/kzg/src/lib.rs#L69-L73) and [`G2`](https://github.com/ArtiomTr/rust-kzg/blob/9482c91f7bb744c8d429bcdc08731282eb1937bc/kzg/src/lib.rs#L99-L101) traits, haven't well-defined endianness. So if you're proposing to rename conversion functions in traits, I absolutely agree - as this will make code more straightforward. However, the functions in `kzg-bench` cannot be changed, simply because the `eip4844` standard doesn't allow to change endianness.

Reply:
Yeah, I think that's a good idea to rename `TFr::from_bytes` to `TFr::from_be_bytes`, although I don't really see a point in implementing `TFr::from_le_bytes` - all `eip4844` functions expect that bytes will be in BE ([see "constants" section "KZG_ENDIANNESS"](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#constants)).

Reply:
Ah right, my mistake, I heard about that switch to BE and thought it applied to all elements though looks like G1/G2 was already BE & only field element was recently changed to BE. I checked arkworks deserialization implementation it seems that G1/G2 already assumed BE format, so all good on that front.

In our case arkworks library Fr serialization assumed input to be LE & that's why that specific test started failing.

Overall I think simplest solution is just to make other teams aware that they should check their Fr from/to bytes/u64 implementations 😄
Or if we decide to change the traits then probably only Fr should have those _le/_be variants.

Anyways, thank you for detailed explanation!

Reply:
Yep, deleted the original comment, as it is a bit irrelevant - `TFr` is not a direct API of `eip4844`, so technically trait could be implemented as you wish. 

I believe we should write additional tests for general traits, so all of them work the same.